### ISSUE TITLE
Issue with portable NVDA on Windows 7

### ISSUE THREAD

**Issue by dparker21** [nvaccess/nvda#8398]
Hi there,

I'm using a portable copy of NVDA 2018.2.2 on a Windows 7 desktop machine for work (I updated from 2018.2.1 yesterday). I am encountering an unpredictable issue with both versions, and I'm not sure what causes it or how to fix it.

I'll be doing something on the machine and NVDA will freeze for a few seconds. Then speech comes back, but other NVDA commands do not work. For example, I cannot use CTRL to silence speech, and I do not have use of the NVDA modifier, whether Caps Lock or Insert (I use laptop keyboard layout). If I eject the flash drive with NVDA and put it in again, it just starts the same behavior again. If I log out of Windows and log in again (a somewhat involved process with this particular system), the speech will start again but still no other functionality. The only fix I know of is a full restart, which wastes time.

Has this been reported before, and if so, is there an easier fix for this? And, more broadly, how does one access the NVDA controls in a situation where the modifier keys aren't working? I have another NVDA copy in a VM for my personal Mac using Parallels Desktop, and my Caps Lock key does not work for the modifier. I have no insert key on that Mac laptop.

Thanks,
Daniel 

### Steps to reproduce:
> A list of the steps you take to demonstrate the problem. 
>
> Example:
>
> 1. Open Chrome
> 2. Browse to www.google.com
> 3. Type "Hello"
> 4. Notice an error sound when enter is pressed.

> Please also remember to attach a zip of any files required to reproduce the issue.

### Expected behavior:
> Tell us what should happen.

### Actual behavior:
> Tell us what happens instead.

### System configuration:
NVDA version:
> Example: next-14027,c80e529f

NVDA Installed or portable:

Other information:
> Example: Running in a VM

Windows version:
> Example: Windows 10 Version 1607 Build 14393.1066

Name and version of other software in use when reproducing the issue:

### Other questions:

Does the issue still occur after restarting your PC?

Have you tried any other versions of NVDA?
> Please list them and the result

**Comment by DrSooom** [comment]
Maybe your flash drive has broken. Backup all data first (!), then format it and run [h2testw](https://www.heise.de/download/product/h2testw-50539) on it. this little bit of software writes and checks the written data on your flash drive. Don't use h2testw on a SSD - HDD is still fine too. Well, or just install NVDA Portable on another flash drive - but do not copy existing one on it, it should be a fresh portable copy of NVDA 2018.2.

**Comment by dparker21** [comment]
Thanks very much, Dr.Sooom, for the response. I apologize for taking so long to get back to it. I decided to just be lazy and install a fresh NVDA copy of NVDA on another flash drive. After a few days, the same thing started happening. Is there something wrong with the way I'm closing out of NVDA at the end of the day? I usually just tell the computer to shut down while NVDA is still running. Is this a problem?

Also, as I asked before, is there a way to force NVDA to restart if you have no control of the NVDA modifier keys?

Thanks,
Daniel

**Comment by Adriani90** [comment]
@dparker21 how is the behavior of NVDA in NVDA 2019.3 Beta 1? Are you still having this issue?

**Comment by dparker21** [comment]
Hi,

I no longer use that PC and Windows 7, and rarely use QVDA these days (I've been using a Mac more often), so I have not had the issue nor an opportunity to test it.

Best, Daniel

**Comment by Adriani90** [comment]
@dparker21 thanks for the swift reply.
Since this issue is unpredictable and cannot be reproduced reliably, I am closing it due to missing tests.
